frontend/catalyst/qjit_device.py

```diff
@@ -218,3 +220,112 @@ def _decomp_to_unitary(self, *_args, **_kwargs):
 
         self.check_validity(expanded_tape.operations, [])
         return expanded_tape
+
+
+class QJITDeviceNewAPI(qml.devices.Device):
+    """QJIT device for the new device API.
+    A device that interfaces the compilation pipeline of Pennylane programs.
+    Args:
+        wires (Shots): the number of wires to initialize the device with
+        shots (int): How many times the circuit should be evaluated (or sampled) to estimate
+            the expectation values. Defaults to ``None`` if not specified. Setting
+            to ``None`` results in computing statistics like expectation values and
+            variances analytically
+        backend_name (str): name of the device from the list of supported and compiled backend
+            devices by the runtime
+        backend_kwargs (Dict(str, AnyType)): An optional dictionary of the device specifications
+    """
+
+    # These must be present even if empty.
+    operations = []
+    observables = []
+
+    operations_supported_by_QIR_runtime = RUNTIME_OPERATIONS
+
+    @staticmethod
+    def _check_mid_circuit_measurement(config):
+        return config["compilation"]["mid_circuit_measurement"]
+
+    @staticmethod
+    def _check_adjoint(config):
+        return config["compilation"]["quantum_adjoint"]
+
+    @staticmethod
+    def _check_quantum_control(config):
+        return config["compilation"]["quantum_control"]
+
+    @staticmethod
+    def _set_supported_operations(config):
+        """Override the set of supported operations."""
+        native_gates = set(config["operators"]["gates"][0]["native"])
+        qir_gates = QJITDeviceNewAPI.operations_supported_by_QIR_runtime
+        QJITDeviceNewAPI.operations = list(native_gates.intersection(qir_gates))
+
+        # These are added unconditionally.
+        QJITDeviceNewAPI.operations += ["Cond", "WhileLoop", "ForLoop"]
+
+        if QJITDeviceNewAPI._check_mid_circuit_measurement(config):  # pragma: no branch
+            QJITDeviceNewAPI.operations += ["MidCircuitMeasure"]
+
+        if QJITDeviceNewAPI._check_adjoint(config):
+            QJITDeviceNewAPI.operations += ["Adjoint"]
+
+        if QJITDeviceNewAPI._check_quantum_control(config):  # pragma: nocover
+            # TODO: Once control is added on the frontend.
```

Comment:
```suggestion
        if QJITDeviceNewAPI._check_quantum_control(config):
```

Reply:
I have just copied what is in main, so it will be removed with Sergei's PR